test/smoke/FirefoxPrivateVPNUITest/FirefoxPrivateVPNUITest/Tests/ConnectTest.cs

```diff
@@ -0,0 +1,126 @@
+// </copyright>
+
+namespace FirefoxPrivateVPNUITest
+{
+    using System;
+    using System.Net;
+    using System.Threading;
+    using FirefoxPrivateVPNUITest.Screens;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using RestSharp;
+
+    /// <summary>
+    /// This is to test connection and disconnection functionality.
+    /// </summary>
+    [TestClass]
+    public class ConnectTest
+    {
+        private FirefoxPrivateVPNSession vpnClient;
+        private BrowserSession browser;
+        private DesktopSession desktop;
+
+        /// <summary>
+        /// Initialize browser, vpn client, and desktop session.
+        /// </summary>
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.browser = new BrowserSession();
+            this.desktop = new DesktopSession();
+            this.vpnClient = new FirefoxPrivateVPNSession();
+        }
+
+        /// <summary>
+        /// Dispose vpn client, browser, and desktop sessions.
+        /// </summary>
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            this.vpnClient.Dispose();
+            this.browser.Dispose();
+            this.desktop.Dispose();
+        }
+
+        /// <summary>
+        /// The test steps.
+        /// </summary>
+        [TestMethod]
+        public void TestConnection()
+        {
+            // Switch to VPN client session
+            this.vpnClient.Session.SwitchTo();
+            LandingScreen landingScreen = new LandingScreen(this.vpnClient.Session);
+            landingScreen.ClickGetStartedButton();
+
+            // User Sign In via web browser
+            UserCommonOperation.UserSignIn(this.vpnClient, this.browser);
+
+            // Main Screen
+            this.vpnClient.Session.SwitchTo();
+            MainScreen mainScreen = new MainScreen(this.vpnClient.Session);
+            Assert.AreEqual("VPN is off", mainScreen.GetTitle());
+            Assert.AreEqual("Turn it on to protect your entire device", mainScreen.GetSubtitle());
+            Assert.IsTrue(mainScreen.GetOffImage().Displayed);
+            Assert.IsFalse(mainScreen.GetOnImage().Displayed);
+
+            // Verify user is not connected to Mullvad VPN
+            IRestResponse response = UserCommonOperation.AmIMullvad();
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.IsTrue(response.Content.Contains("You are not connected to Mullvad"));
+
+            // Click VPN switch toggle and turn on VPN
+            mainScreen.ToggleVPNSwitch();
+            Assert.AreEqual("Connecting...", mainScreen.GetTitle());
+            Assert.AreEqual("You will be protected shortly", mainScreen.GetSubtitle());
+            Assert.IsTrue(mainScreen.GetOnImage().Displayed);
+            Assert.IsFalse(mainScreen.GetOffImage().Displayed);
+
+            // Verify the windows notification
+            Thread.Sleep(TimeSpan.FromSeconds(2));
```

Comment:
Not guaranteed.

Reply:
It will auto retry 3 times to check the windows notification either from notification toast or action center.